packages/utils/core-utils/src/package-manager.ts

```diff
@@ -0,0 +1,103 @@
+import { CrossSpawnArgs, CrossSpawnOptions, spawn } from '@malept/cross-spawn-promise';
+import chalk from 'chalk';
+import findUp from 'find-up';
+import logSymbols from 'log-symbols';
+
+export type SupportedPackageManager = 'yarn' | 'npm' | 'pnpm';
+export type PMDetails = { executable: SupportedPackageManager; version?: string; install: string; dev: string; exact: string };
+
+const MANAGERS: Record<SupportedPackageManager, PMDetails> = {
+  yarn: {
+    executable: 'yarn',
+    install: 'add',
+    dev: '--dev',
+    exact: '--exact',
+  },
+  npm: {
+    executable: 'npm',
+    install: 'install',
+    dev: '--save-dev',
+    exact: '--save-exact',
+  },
+  pnpm: {
+    executable: 'pnpm',
+    install: 'add',
+    dev: '--save-dev',
+    exact: '--save-exact',
+  },
+};
+
+const PM_FROM_LOCKFILE: Record<string, SupportedPackageManager> = {
+  'package-lock.json': 'npm',
+  'yarn.lock': 'yarn',
+  'pnpm-lock.yaml': 'pnpm',
+};
+
+/**
+ * Parses the `npm_config_user_agent` environment variable and returns its name and version.
+ *
+ * Code taken from {@link https://github.com/zkochan/packages/tree/main/which-pm-runs/ | which-pm-runs}.
+ */
+function pmFromUserAgent() {
+  const userAgent = process.env.npm_config_user_agent;
+  if (!userAgent) {
+    return undefined;
+  }
+  const pmSpec = userAgent.split(' ')[0];
```

Comment:
nit(Optional): Slight performance improvement :)

```suggestion
  const pmSpec = userAgent.split(' ', 1)[0];
```